### ISSUE TITLE
Hangs after session_start() in `includes/session.php` in PHP 5.3.1

### ISSUE THREAD

**Issue by apmuthu** [timschofield/webERP-svn#17]
Cannot login using PHP 5.3.1 in WinXPSP3.
Hangs at https://github.com/timschofield/webERP-svn/blob/master/includes/session.php#L39

**Comment by apmuthu** [comment]
admin user password `weberp` hash in `www_users` table set to:
````
$2n/2fEu2eudU
````
This is for PHP < 5.4

**Comment by TurboPT** [comment]
As you mentioned Windows XP SP3, what browser and version is in use on that system?

**Comment by apmuthu** [comment]
FireFox v52.2 ESR.

Also during installation, after inculcating the Fix for #16, only the `weberpdemo` company gets installed and the `weberp` company's 155 db tables do not get created (XAMPP v1.7.3).

What other data / code elements will change if a company is migrated from one installation to another - other than the config.php entry and some records in the `config` table?

Please note that the entries in the `config` table for the `weberpdemo` company refer to `companies/test/...` instead of `companies/weberpdemo/....` that need to be corrected in the `demo.sql` file and possibly in the code as well.

**Comment by TurboPT** [comment]
Do the tables fail to install because of timeout, or some other issue with XAMPP?

For the original session_start(), are you sure it is that function, and not the line before that point?

Which PHP and MySQL versions are in use with what you have?

**Comment by apmuthu** [comment]
If timeout be an issue, some sort of progress bar would expose it.

The fact is that no tables are created at all during the time the browser is waiting for data from the server, when monitored from a separate MySQL client. The XAMPP 1.7.3 sports PHP v5.3.1 and MySQL v5.1.41. By placing `echo` and an `exit` statements first before and then after the `session_start()` statement, the former displayed whilst the latter hung.

**Comment by TurboPT** [comment]
What is the **session.save_handler** value in php.ini with XAMPP?

**Comment by TurboPT** [comment]
That is the right setting. I'm trying to think of other things, and I'm running short on ideas.

What is your path to XAMPP?

**Comment by apmuthu** [comment]
There is no UAC in WinXP. Tried `c:\XAMPP`, `d:\xampp`, `d:\servers\xampp` - all same result.

**Comment by TurboPT** [comment]
True, I forgot about XP and (the lack of) UAC.

Are there any potential clues in the apace log, or possibly the Windows admin event logs?

**Comment by apmuthu** [comment]
No clues anywhere in apache / application error logs.
It is necessary to run WebERP in such base level versions to provide confidence to those whose cloud instances fail and need a local instance where Windows of every hue is still popular and cannot be wished away any time soon.
What is the manual install sequence other than to
1.  create a database with same name as the company,  
2. copy of the `weberpdemo` folder,  
3. manual run of the `default.sql` on the company's database,  
4. replacing the hashed password in the user table to degrade to what is available in the PHP version used and
5. make an entry in the `config` file for the login dropdown box?

**Comment by apmuthu** [comment]
My existing settings are:
````
error_reporting = E_ALL & ~E_NOTICE & ~E_DEPRECATED
display_errors = On
display_startup_errors = On
log_errors = Off
magic_quotes_gpc = Off
magic_quotes_runtime = Off
register_globals = Off
````
Even after turning on the `log_errors`, still no joy in the apache error log.
The `gettext` extension has been tested being turned on and commented out to no avail.

**Comment by apmuthu** [comment]
SQL fixes in #28 should fix this issue too. Will test and revert.

**Comment by apmuthu** [comment]
No joy.

The following have been discerned:
After Line 466 in `install/index.php`, the `$Db` is undefined. The value of the `$_POST` and other variables there are:
````
$NewSQLFile = '../sql/mysql/country_sql/default.sql';
$_POST = Array ( 
[CompanyName] => weberp
[CountrySQL] => default.sql
[TimeZone] => Asia/Kolkata
[Email] => admin@example.com
[webERPPassword] => mypwd
[PasswordConfirm] => mypwd
[HostName] => localhost
[UserName] => root
[DatabaseName] => weberp
[Password] =>
[MysqlExt] =>
[UserLanguage] => en_GB.utf8
[MAX_FILE_SIZE] => 10240
[Install] =>
[Database] => weberp
) ;
````
Note that there is no place in the install form for choosing `MysqlExt` as `mysql` or `mysqli`.

**Comment by apmuthu** [comment]
The function that fails is:
````
PopulateSQLData()
````
This is since the `$Db` value does not get passed onto the `$db` argument in the function.

When only a single database is chosen to be created during install, the values of the arguments passed to the above function call in [Line 467 of `install/index.php`](https://github.com/webERP-team/webERP/blob/master/install/index.php#L467) are:
````
$Demo: bool(false)
$Db: object(mysqli)#1 (17) { 
  ["affected_rows"]=> int(1) 
  ["client_info"]=> string(6) "5.1.41" 
  ["client_version"]=> int(50141) 
  ["connect_errno"]=> int(0) 
  ["connect_error"]=> NULL 
  ["errno"]=> int(0) 
  ["error"]=> string(0) "" 
  ["field_count"]=> int(0) 
  ["host_info"]=> string(20) "localhost via TCP/IP" 
  ["info"]=> NULL 
  ["insert_id"]=> int(0) 
  ["server_info"]=> string(6) "5.1.41" 
  ["server_version"]=> int(50141) 
  ["sqlstate"]=> string(5) "00000" 
  ["protocol_version"]=> int(10)
  ["thread_id"]=> int(72)
  ["warning_count"]=> int(1)
}
$DBConnectType: string(6) "mysqli"
$DatabaseName: string(6) "weberp"
````
Furthermore, there is no check for successful completion on return from this function before the next one proceeds to update with the user preference data.

Attempting to single-step into the `function PopulateSQLData()` shows that the `$Db` object does not get passed on inside the function as `$db` properly. Does this need to be passed by reference in older PHP versions?

**Comment by apmuthu** [comment]
Further fixes [here](https://github.com/timschofield/webERP-svn/issues/28#issuecomment-378212644) result in error-free installation.

**Comment by TurboPT** [comment]
Tim, can this issue be closed, fixed by #28 ?
